File: FragmentDemo/app/src/main/java/com/geng/fragmentdemo/Main2Activity.java
package com.geng.fragmentdemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class Main2Activity extends AppCompatActivity {

    public static final int RESULT_CODE = 200;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
    }


    public void backTo(View view) {
        Intent intent = new Intent();
        intent.putExtra("result", "我是main 2");
        setResult(RESULT_CODE, intent);
        finish();
    }
}

File: ScrollScrollViewListView/app/src/main/java/com/gjr/scroll_scrollviewlistview/Main2Activity.kt
package com.gjr.scroll_scrollviewlistview

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.widget.AbsListView
import kotlinx.android.synthetic.main.activity_main2.*

class Main2Activity : AppCompatActivity() {

    var page: Int = 1
    val adapter = MyAdapter2(DataBuilder.buildData(page))
//    val adapter = MyAdapter(this,DataBuilder.buildData(page))

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        initData()
    }


    private fun initData() {
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)


        // 为了减少RecycleView滑动时的卡顿（并不如丝般顺滑）
        recyclerView.isNestedScrollingEnabled = false // 默认是 true， 在构造方法中



//        listView.adapter = adapter
//        listView.setOnScrollListener(object : AbsListView.OnScrollListener {
//            override fun onScrollStateChanged(view: AbsListView, scrollState: Int) {
//                if (scrollState == AbsListView.OnScrollListener.SCROLL_STATE_IDLE
//                        && (view.lastVisiblePosition == adapter.data.size - 1)) {
//                    adapter.data.addAll(DataBuilder.buildData(++page))
//                    adapter.notifyDataSetChanged()
//                }
//            }
//
//            override fun onScroll(view: AbsListView, firstVisibleItem: Int, visibleItemCount: Int, totalItemCount: Int) {
//            }
//        })
    }
}
